future-apps/payroll/contracts/Payroll.sol

```diff
@@ -250,121 +324,159 @@ contract Payroll is AragonApp { //, IForwarder { // makes coverage crash (remove
-    function changeAddressByEmployee(address newAddress) external {
+    function changeAddressByEmployee(address newAddress) isInitialized employeeMatches external {
         // check that account doesn't exist
         require(employeeIds[newAddress] == 0);
         // check it's non-null address
         require(newAddress != address(0));
-        // check that employee exists (and matches)
+
         uint128 employeeId = employeeIds[msg.sender];
         Employee storage employee = employees[employeeId];
-        // check that employee exists (and matches)
-        require(employee.accountAddress == msg.sender);
 
+        emit ChangeAddressByEmployee(employeeId, employee.accountAddress, newAddress);
         employee.accountAddress = newAddress;
         employeeIds[newAddress] = employeeId;
-        employeeIds[msg.sender] = 0;
+        delete employeeIds[msg.sender];
     }
 
     /**
-     * @dev Return all important info too through employees mapping
+     * @dev Return all Employee's important info
+     * @notice Return all Employee's important info
+     * @param accountAddress Employee's address to receive payments
+     * @return Employee's identifier
+     * @return Employee's annual salary, per second in denomination Token
+     * @return Employee's accrued value
+     * @return Employee's name
+     * @return Employee's last payment received date
+     * @return Employee's termination date (max uint64 if none)
+     * @return Bool indicating if employee is terminated
+     */
+    function getEmployeeByAddress(address accountAddress)
+        external
+        view
+        returns (
+            uint128 employeeId,
+            uint256 denominationSalary,
+            uint256 accruedValue,
+            string name,
+            uint64 lastPayroll,
+            uint64 endDate,
+            bool terminated
+        )
+    {
+        employeeId = employeeIds[accountAddress];
+
+        Employee memory employee = employees[employeeId];
+
+        denominationSalary = employee.denominationTokenSalary;
+        accruedValue = employee.accruedValue;
+        name = employee.name;
+        lastPayroll = employee.lastPayroll;
+        endDate = employee.endDate;
+        terminated = employee.terminated;
+    }
+
+    /**
+     * @dev Return all Employee's important info
      * @notice Return all Employee's important info
      * @param employeeId Employee's identifier
      * @return Employee's address to receive payments
-     * @return Employee's annual salary
+     * @return Employee's annual salary, per second in denomination Token
+     * @return Employee's accrued value
      * @return Employee's name
-     * @return Employee's last call to payment distribution date
      * @return Employee's last payment received date
+     * @return Employee's termination date (max uint64 if none)
+     * @return Bool indicating if employee is terminated
      */
     function getEmployee(uint128 employeeId)
-        external
+        public
```

Comment:
I don't remember why this change. `getEmployeeByAddress` is kept as `external`.

Reply:
Would you do the same with `isTokenAllowed`?